src/game.cpp

```diff
@@ -6271,6 +6271,20 @@ bool game::spawn_hallucination()
     }
 }
 
+bool game::spawn_hallucination( const tripoint &p )
```

Comment:
This looks like a straight copy of the existing `spawn_hallucination` function. Don't so much nearly identicall code. Change the other function to just create the point where to spawn and forward it to this new function.